src/collectors/fio/fio.py

```diff
@@ -0,0 +1,121 @@
+# coding=utf-8
+
+"""
+The FioCollector collects results from fio runs
+
+#### Dependencies
+
+ * fio
+
+"""
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import re
+import subprocess
+
+import diamond.collector
+
+
+def flatten_dictionary(input, sep='.', prefix=None):
+    """Produces iterator of pairs where the first value is
+    the joined key names and the second value is the value
+    associated with the lowest level key. For example::
+
+      {'a': {'b': 10},
+       'c': 20,
+       }
+
+    produces::
+
+      [('a.b', 10), ('c', 20)]
+    """
+
+    for key, value in sorted(input.items()):
+        if isinstance(value, dict):
+            for subkey, subvalue in flatten_dictionary(value):
+                yield key + "." + subkey, subvalue
+        else:
+                yield key, value
+
+
+class FioCollector(diamond.collector.Collector):
+    def get_default_config_help(self):
+        config_help = super(FioCollector, self).get_default_config_help()
+        config_help.update({
+            'ioengine': 'default: libaio',
+            'iodepth': 'default: 4',
+            'rw': 'default: readwrite',
+            'bs': 'default: 32k',
+            'direct': 'default: 0',
+            'size': 'default: 100M',
+            'directory': 'default: /tmp'
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+
+        config = super(FioCollector, self).get_default_config()
+        config.update({
+            'ioengine': 'libaio',
+            'iodepth': '1',
+            'rw': 'readwrite',
+            'bs': '32k',
+            'direct': '0',
+            'size': '100M',
+            'directory': '/tmp',
+        })
+        return config
+
+    def _get_stats_from_fio(self):
+        """Return the parsed JSON fio output
+        """
+
+        try:
+            json_blob = subprocess.check_output(
+                ['/usr/bin/fio',
```

Comment:
Typically you wouldn't want to hardcode this location either, since someone may have it installed into `/usr/local/bin` instead, or any number of other places. So I'd move this to a configurable option, and have this as a default.

Reply:
This still needs to be dealt with.